static/app/views/settings/components/teamSelect.tsx

```diff
@@ -19,6 +20,12 @@ import {Organization, Team} from 'sentry/types';
 import useTeams from 'sentry/utils/useTeams';
 
 type Props = {
+  /**
+   * Is this component for a project
+   * and should adding a team be disabled based
+   * on whether the team is idpProvisioned
+   */
+  addingTeamToProject: boolean;
```

Comment:
For boolean props, it's generally good practice to start it with `is` or `has`. 

However, there's precedence on the `OrganizationMemberDetail` page where the `OrganizationRoleSelect` is using `enforceAllowed` and `enforceRetired` (see [link](https://github.com/getsentry/sentry/blob/2eac063cf8989dfc0791ca8b12c78575631b22ac/static/app/views/settings/organizationMembers/organizationMemberDetail.tsx#L356)). 

As such, a better name for this property would be `enforceIdpProvisioned`, then on the Teams Page, set it to `false`.